spec/flipper/api/action_spec.rb

```diff
@@ -25,35 +25,72 @@ def delete
     end
   }
 
-  it "won't run method that isn't whitelisted" do
-    fake_request = Struct.new(:request_method, :env, :session).new("NOOOOPE", {}, {})
-    action = action_subclass.new(flipper, fake_request)
-    expect {
-      action.run
-    }.to raise_error(Flipper::Api::RequestMethodNotSupported)
-  end
+  describe 'https verbs' do
 
-  it "will run get" do
-    fake_request = Struct.new(:request_method, :env, :session).new("GET", {}, {})
-    action = action_subclass.new(flipper, fake_request)
-    expect(action.run).to eq([200, {}, "get"])
-  end
+    it "won't run method that isn't whitelisted" do
+      fake_request = Struct.new(:request_method, :env, :session).new("NOOOOPE", {}, {})
+      action = action_subclass.new(flipper, fake_request)
+      expect {
+        action.run
+      }.to raise_error(Flipper::Api::RequestMethodNotSupported)
+    end
 
-  it "will run post" do
-    fake_request = Struct.new(:request_method, :env, :session).new("POST", {}, {})
-    action = action_subclass.new(flipper, fake_request)
-    expect(action.run).to eq([200, {}, "post"])
-  end
+    it "will run get" do
+      fake_request = Struct.new(:request_method, :env, :session).new("GET", {}, {})
+      action = action_subclass.new(flipper, fake_request)
+      expect(action.run).to eq([200, {}, "get"])
+    end
 
-  it "will run put" do
-    fake_request = Struct.new(:request_method, :env, :session).new("PUT", {}, {})
-    action = action_subclass.new(flipper, fake_request)
-    expect(action.run).to eq([200, {}, "put"])
+    it "will run post" do
+      fake_request = Struct.new(:request_method, :env, :session).new("POST", {}, {})
+      action = action_subclass.new(flipper, fake_request)
+      expect(action.run).to eq([200, {}, "post"])
+    end
+
+    it "will run put" do
+      fake_request = Struct.new(:request_method, :env, :session).new("PUT", {}, {})
+      action = action_subclass.new(flipper, fake_request)
+      expect(action.run).to eq([200, {}, "put"])
+    end
+
+    it "will run delete" do
+      fake_request = Struct.new(:request_method, :env, :session).new("DELETE", {}, {})
+      action = action_subclass.new(flipper, fake_request)
+      expect(action.run).to eq([200, {}, "delete"])
+    end
   end
 
-  it "will run delete" do
-    fake_request = Struct.new(:request_method, :env, :session).new("DELETE", {}, {})
-    action = action_subclass.new(flipper, fake_request)
-    expect(action.run).to eq([200, {}, "delete"])
+  describe '#json_error_response' do
+    describe ":feature_not_found" do
+
+      it 'locates and serializes error correctly' do
+        action = action_subclass.new({}, {})
+        response = catch(:halt) do
+          action.json_error_response(:feature_not_found, 404)
+        end
+        expect(response).to eq([404, {"Content-Type"=>"application/json"}, ["{\"code\":1,\"message\":\"Feature not found.\",\"more_info\":\"\"}"]])
+      end
```

Comment:
I am ok with this. Thanks for adding the tests! 

If I were to change anything, I might assert each part separately. I find separate assertions easier as they more directly point to the assertion failure. Instead of searching through a big object for a differences you can just look at simple assertions.

```ruby
status, headers, body = [...]
body_hash = JSON.parse(body[0])
assert_equal 404, status
assert_equal "application/json", headers["Content-Type"]
assert_equal 1, body_hash["code"]
assert_equal "Feature not found.", body_hash["message"]
assert_equal "", body_hash["more_info"]
```